### ISSUE TITLE
ExceededBatchRecordSizeException was swallowed after fail to append recode in BufferedTaskResultBuilder

### ISSUE THREAD

**Issue by weiye-Lee** [camunda/zeebe#14058]
**Describe the bug**

in the following code, the eigher.left Exception will be swallowed, it will cause the exception throwed until the actually run tryWrite().

by the way😅, [Help Needed](https://forum.camunda.io/t/writing-of-scheduled-taskresult-failed/47247)

DirectTaskResultBuilder.java
```java
  @Override
  public boolean appendCommandRecord(
      final long key, final Intent intent, final UnifiedRecordValue value) {

    final ValueType valueType = TYPE_REGISTRY.get(value.getClass());
    if (valueType == null) {
      // usually happens when the record is not registered at the TypedStreamEnvironment
      throw new IllegalStateException("Missing value type mapping for record: " + value.getClass());
    }

    final var either = mutableRecordBatch.appendRecord(
        key, -1, RecordType.COMMAND, intent, RejectionType.NULL_VAL, "", valueType, value);

    return either.isRight();
  }
````

**To Reproduce**

👀

**Expected behavior**

Expected to catch the Either left and throw ExceededBatchRecordSizeException.

RecordBatch.java#appendRecord
```java
    if (!recordBatchSizePredicate.test(recordBatchEntries.size() + 1, batchSize + entryLength)) {
      return Either.left(
          new ExceededBatchRecordSizeException(
              recordBatchEntry, entryLength, recordBatchEntries.size(), batchSize));
    }
```

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
dev-zeebe-1  | 2023-08-30 08:48:03.185 [Broker-0-StreamProcessor-3] [Broker-0-zb-actors-0] WARN 
dev-zeebe-1  |       io.camunda.zeebe.broker.logstreams - Writing of scheduled TaskResult failed!
dev-zeebe-1  | java.lang.IllegalArgumentException: Expected to claim segment of size 4270264, but can't claim more than 4194304 bytes.
dev-zeebe-1  | 	at io.camunda.zeebe.dispatcher.Dispatcher.offer(Dispatcher.java:207) ~[zeebe-dispatcher-8.1.5.jar:8.1.5]
dev-zeebe-1  | 	at io.camunda.zeebe.dispatcher.Dispatcher.claimFragmentBatch(Dispatcher.java:164) ~[zeebe-dispatcher-8.1.5.jar:8.1.5]
dev-zeebe-1  | 	at io.camunda.zeebe.logstreams.impl.log.LogStreamBatchWriterImpl.claimBatchForEvents(LogStreamBatchWriterImpl.java:235) ~[zeebe-logstreams-8.1.5.jar:8.1.5]
dev-zeebe-1  | 	at io.camunda.zeebe.logstreams.impl.log.LogStreamBatchWriterImpl.tryWrite(LogStreamBatchWriterImpl.java:212) ~[zeebe-logstreams-8.1.5.jar:8.1.5]
dev-zeebe-1  | 	at io.camunda.zeebe.streamprocessor.ProcessingScheduleServiceImpl.lambda$toRunnable$5(ProcessingScheduleServiceImpl.java:165) ~[zeebe-workflow-engine-8.1.5.jar:8.1.5]
dev-zeebe-1  | 	at io.camunda.zeebe.scheduler.retry.ActorRetryMechanism.run(ActorRetryMechanism.java:28) ~[zeebe-scheduler-8.1.5.jar:8.1.5]
dev-zeebe-1  | 	at io.camunda.zeebe.scheduler.retry.AbortableRetryStrategy.run(AbortableRetryStrategy.java:45) ~[zeebe-scheduler-8.1.5.jar:8.1.5]
dev-zeebe-1  | 	at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:92) ~[zeebe-scheduler-8.1.5.jar:8.1.5]
dev-zeebe-1  | 	at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:45) ~[zeebe-scheduler-8.1.5.jar:8.1.5]
dev-zeebe-1  | 	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:119) ~[zeebe-scheduler-8.1.5.jar:8.1.5]
dev-zeebe-1  | 	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:106) ~[zeebe-scheduler-8.1.5.jar:8.1.5]
dev-zeebe-1  | 	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:87) ~[zeebe-scheduler-8.1.5.jar:8.1.5]
dev-zeebe-1  | 	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:198) ~[zeebe-scheduler-8.1.5.jar:8.1.5]
```

</p>
</details>

**Environment:**

- Zeebe Version: release-8.1.15

**Comment by korthout** [comment]
@megglos This issue is related to the Stream Platform, removing it from ZPA team's board

**Comment by weiye-Lee** [comment]
Well, i will close it cause i figured out the reason of doesn't catch the `Either.left`, it just try to append the max record.It's just a flag of couldn't append more recode when return an Either.left.

And my logstrack is caused by the following code which fixed in version of 8.2.0 , the test method has a slight 
deviation  from the actual tryWrite() behavior."

RecordBatch 8.1.14
```java
    // Can't use RecordBatchEntry.getLength here, as it includes the key and source index length
    // The to be called recordBatchSizePredicate expects only metadata and value length to be passed
    final var entryLength =
        recordBatchEntry.getMetadataLength() + recordBatchEntry.getValueLength();

    if (!recordBatchSizePredicate.test(recordBatchEntries.size() + 1, batchSize + entryLength)) {
      return Either.left(
          new ExceededBatchRecordSizeException(
              recordBatchEntry, entryLength, recordBatchEntries.size(), batchSize));
    }
```

(RecordBatch 8.2.0)[https://github.com/camunda/zeebe/blob/8.2.0/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/records/RecordBatch.java]
```java
    final var entryLength = recordBatchEntry.getLength();

    if (!recordBatchSizePredicate.test(recordBatchEntries.size() + 1, batchSize + entryLength)) {
      return Either.left(
          new ExceededBatchRecordSizeException(
              recordBatchEntry, entryLength, recordBatchEntries.size(), batchSize));
    }
```
